presto-pinot-toolkit/src/main/java/com/facebook/presto/pinot/PinotPushdownUtils.java

```diff
@@ -147,10 +154,25 @@ public ExpressionType getExpressionType()
             if (agg != null) {
                 if (agg.getFilter().isPresent()
                         || agg.isDistinct()
-                        || agg.getOrderBy().isPresent()
-                        || agg.getMask().isPresent()) {
+                        || agg.getOrderBy().isPresent()) {
+                    throw new PinotException(PINOT_UNSUPPORTED_EXPRESSION, Optional.empty(), "Unsupported aggregation node " + aggregationNode);
+                }
+                if (agg.getMask().isPresent()) {
+                    /**
```

Comment:
I am wondering if this comment can be simplified to saying that: This block handles the case when a distinct aggregation is present in addition to another aggregation function. And the example need not contain a subquery ?

Reply:
will do